File: Inventory Management System/src/main/java/com/example/inventorymanagementsystem/DTO/ProductUserResponse.java
package com.example.inventorymanagementsystem.DTO;

import java.math.BigDecimal;
import java.time.Instant;

public record ProductUserResponse(
        Long id,
        String sku,
        String name,
        Long categoryId,
        String categoryName,
        Integer quantityOnHand,
        BigDecimal unitPrice,
        String status,
        String description,
        Instant createdAt,
        Instant updatedAt
) {
}

File: FrontEnd/frontend-app/src/app/Model/product-user-response.ts
export interface ProductUserResponse {
     id: number;
  name: string;
  description: string;
  sku: string;

  categoryId: number;
  categoryName: string;

  unitPrice: number;

  quantityOnHand: number;
  reorderLevel: number;

  status: string;

  createdAt: string;
  updatedAt: string;

  tempQty?: number;
}

